### PR TITLE
Add Experiment Creation Form with React Hook Form

### PR DESCRIPTION
**Pull request by dhurlburtusa**
A give-up implementation of the Experiment creation form with minimal UI layout for easier comparison.

## How has this been tested?

This is not meant to be merged. It is to be compared with another WIP PR that will be committed today or tomorrow.

Also, note that this is not intended to be a polished product. There may be some extra stuff laying around that is meant to spark a conversation or just because I thought it would be better to get feedback before spending too much time to implementing something we won't need.

### CODE REVIEW

File: pages/new.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import debugFactory from 'debug'
```
Comment (yanirs) [inline pages/new.tsx:1]: Let's put this under `/experiments/new`.

File: package.json [hunk 2]
```diff
@@ -14,16 +14,22 @@
     "node": ">=10.18.1"
   },
   "dependencies": {
+    "classnames": "^2.2.6",
     "date-fns": "^2.13.0",
     "debug": "^4.1.1",
     "isomorphic-unfetch": "^3.0.0",
+    "json-stringify-safe": "^5.0.1",
```
Comment (dhurlburtusa) [inline package.json:21]: Don't appear to be used in this PR.

File: package.json [hunk 3]
```diff
@@ -39,14 +45,19 @@
     "@storybook/react": "^5.3.18",
     "@testing-library/jest-dom": "^5.5.0",
     "@testing-library/react": "^10.0.3",
+    "@types/classnames": "^2.2.10",
     "@types/debug": "^4.1.5",
     "@types/expect-puppeteer": "^4.4.1",
     "@types/jest": "^25.2.1",
     "@types/jest-environment-puppeteer": "^4.3.1",
+    "@types/json-stringify-safe": "^5.0.0",
```
Comment (dhurlburtusa) [inline package.json:53]: Don't appear to be used in this PR.

File: pages/new.tsx (hunk 1 above)
Comment (dhurlburtusa) [reply pages/new.tsx:1]: Cool. That would be my preference.

@yanirs Do we still want the experiment list at `/` or should we move it to `/experiments`? If so, what do we want at `/`? 

In an interface like Abacus, it is not uncommon to have a dashboard at `/` after the user authenticates. I don't think there is a need at the moment for a dashboard but maybe in the future.

Comment (yanirs) [reply pages/new.tsx:1]: I think we can keep the experiment list under `/` for now, but I agree that having a dashboard there with the experiment list under `/experiments` makes more sense once we have more exciting stuff to show. :slightly_smiling_face: Can do that as part of the polishing iterations.

File: components/ExperimentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+import React from 'react'
+import ReactDatePicker from 'react-datepicker'
+import { Controller, useFieldArray, useForm } from 'react-hook-form'
+import ReactSelect from 'react-select'
+
+import ExperimentsApi from '@/api/ExperimentsApi'
+
+import { ExperimentFull, Platform } from '@/models/index'
+
+const SNAKE_CASE_RE = /^[a-z][a-z0-9_]*[a-z0-9]$/
```
Comment (withinboredom) [inline components/ExperimentForm.tsx:16]: Regex's are starting to become scattered all over the code base without much (if any) documentation which requires parsing regex's to understand what they do, but no idea what they should do. Can we start a `regex.ts` file that we can then import these from.

Comment (dhurlburtusa) [reply components/ExperimentForm.tsx:16]: I'll try to remember to centralize our regexp in a module when doing the React Final Form code. Please remind me if I forget.

**Comment by dhurlburtusa** [comment]
We have decided to say goodbye to React Hook Form and use React Final Form.

**Pull request closed without merging**